system/jlib/jstring.cpp

```diff
@@ -1063,13 +1063,48 @@ StringBuffer & StringBuffer::replaceString(const char* oldStr, const char* newSt
     if (curLen && oldStr)
     {
         size_t oldlen = strlen(oldStr);
-        if (oldlen > curLen)
```

Comment:
I prefer the guard condition.  It simplifies the rest of the function, and makes the function easier to read - because it is clear that this is a case that needs no further processing.